src/lib.rs

```diff
@@ -305,19 +313,46 @@ pub fn run_with(cmd_args: Option<&[&str]>) -> Result<(), String> {
         );
     }
 
+    let paths = fs::read_dir(&args.nodes_dir)
+        .map_err(|_| "Could not read existing testnet log dir".to_string())?;
+
+    let existing_nodes_count = paths
+        .collect::<Result<Vec<_>, io::Error>>()
+        .map_err(|_| "Error collecting testnet log dir".to_string())?
+        .len();
+
+    println!("existing nodes: {:?}", existing_nodes_count);
+
+    // either we have genesis only, or all existing ndoes
+    let start = if existing_nodes_count == 0 {
```

Comment:
Should this be `== 1`? Because the genesis node would be having its directory.

Reply:
hmm, yeh this should not be hittable... 🤔  let me try removing it